airsonic-main/src/main/java/org/airsonic/player/service/search/AnalyzerFactory.java

```diff
@@ -1,121 +1,182 @@
+     * 
+     * The tag parser performs special character conversion
+     * when converting input values ​​from a file.
+     * Therefore, the domain value may be different from the original value.
+     * This filter allows searching by user readable value (file tag value).
+     * 
+     * @see org.jaudiotagger.tag.id3.framebody.FrameBodyTCON#convertID3v23GenreToGeneric
+     * (TCON stands for Genre with ID3 v2.3-v2.4)
+     * Such processing exists because brackets in the Gener string have a special meaning.
+     */
+    private void addTokenFilterForTokenToDomainValue (Builder builder)  throws IOException{
+        builder
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\(", "replacement", "", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\)$", "replacement", "", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\)", "replacement", " ", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\{\\}", "replacement", "\\{ \\}", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\[\\]", "replacement", "\\[ \\]", "replace", "all");
+    }
+
+    private Builder createDefaultAnalyzerBuilder() throws IOException {
+        Builder builder = CustomAnalyzer.builder()
+                .withTokenizer(StandardTokenizerFactory.class)
+                .addTokenFilter(CJKWidthFilterFactory.class)
+                .addTokenFilter(ASCIIFoldingFilterFactory.class, "preserveOriginal", "false")
+                .addTokenFilter(LowerCaseFilterFactory.class)
+                .addTokenFilter(StopFilterFactory.class)
+                .addTokenFilter(EnglishPossessiveFilterFactory.class);
+        addTokenFilterForUnderscoreRemovalAroundToken(builder);
+        return builder;
+    }
+
+    private Builder createKeywordAnalyzerBuilder() throws IOException {
+        return CustomAnalyzer.builder()
+                .withTokenizer(KeywordTokenizerFactory.class);
+    }
+
+    private Builder createGenreAnalyzerBuilder() throws IOException {
+        Builder builder = createKeywordAnalyzerBuilder();
+        addTokenFilterForTokenToDomainValue(builder);
+        return builder;
+    }
+
+    /**
+     * Returns the Analyzer to use when generating the index.
+     * 
+     * Whether this analyzer is applied to input values ​​depends on
+     * the definition of the document's fields.
+     * 
+     * @return analyzer for index
+     * @see DocumentFactory
+     */
+    public Analyzer getAnalyzer() {
+        if (isEmpty(analyzer)) {
+            try {
+                analyzer = createDefaultAnalyzerBuilder().build();
+            } catch (IOException e) {
+                LOG.error("Error when initializing Analyzer.", e);
```

Comment:
Why discard the error here? Maybe wrap it with a runtime exception and let it bubble up?

Reply:
I revised that according to your advice.